tests/sample/cgroup_mt_connect4.c

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+// Whenever this sample program changes, bpf2c_tests will fail unless the
+// expected files in tests\bpf2c_tests\expected are updated. The following
+// script can be used to regenerate the expected files:
+//     generate_expected_bpf2c_output.ps1
+//
+// Usage:
+// .\scripts\generate_expected_bpf2c_output.ps1 <build_output_path>
+// Example:
+// .\scripts\generate_expected_bpf2c_output.ps1 .\x64\Debug\
+
+#include "bpf_endian.h"
+#include "bpf_helpers.h"
+#include "net/ip.h"
+#include "socket_tests_common.h"
+
+const uint16_t remote_port = SOCKET_TEST_PORT;
+const uint16_t redirect_offset = 1000;
+
+SEC("cgroup/connect4")
```

Comment:
IPv6 works just fine.  It's just that we have an existing issue where we cannot have an IPv4 (CGROUP/CONNECT4) _and_ an IPv6 (CGROUPT/CONNECT6) program attached at the same time (Issue #2351).  This would have worked equally well with IPv6. I just _happened_ to pick IPv4 here.

Reply:
Added IPv6 sample program and test updated to use new program.